autoscale_cloudroast/test_repo/autoscale/system/integration/test_system_integration_rcv3.py

```diff
@@ -650,6 +665,116 @@ def test_scale_up_down_on_rcv3_and_clb(self):
                           'count of {1}'.format(num_clb_nodes_after_scale,
                                                 self.min_servers))
 
+    @tags(speed='slow', type='rcv3', rcv3_mimic='pass')
+    def test_scale_down_after_manual_remove_node(self):
+        """
+        Verify that the RCV3 group can successfully scale down after a server
+        has been removed from the loadbalancer pool outside of autoscale.
+        """
+
+        # Capture a list of the node_ids of all nodes on the pool before doing
+        # anything.
+        initial_node_list = [n.id for n in self.rcv3_client.get_nodes_on_pool(
+            self.pool.id).entity.nodes]
+
+        # Create a group with an RCV3 loadbalancer pool and a minimum number
+        # of servers
+
+        lb_pools = [{'loadBalancerId': self.pool.id, 'type': 'RackConnectV3'}]
+
+        pool_group_resp = self._create_rcv3_group(
+            lb_list=lb_pools, group_min=0,
+            network_list=[self.rackconnect_network])
+        pool_group = pool_group_resp.entity
+
+        # Construct a policy to scale up servers
+        server_count_before_autoscaling = self._get_node_counts_on_pool(
+            self.pool.id)['cloud_servers']
+
+        scale_amt = 2
+        policy_up_data = {'change': scale_amt, 'cooldown': 0}
+        expected_rcv3_server_count = (server_count_before_autoscaling
+                                      + scale_amt)
+        as_server_count = scale_amt
+
+        # Create the policy and execute it immediately
+        self.asb.create_policy_webhook(pool_group.id,
+                                       policy_up_data,
+                                       execute_policy=True)
+
+        server_list_after_up, nodes_after = self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count,
+            as_server_count)
+
+        # Remove one of the servers from the LB pool
+        self.rcv3_client.remove_server_from_pool(self.pool.id,
+                                                 server_list_after_up[0])
+
+        post_del_servers, post_del_nodes = self._verify_rcv3_group_state(
```

Comment:
In a perfect world it might not be, but remove_server_from_pool simply calls the rackconnect API and I think it is good practice to confirm that the requested action was actually carried out.